modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C

```diff
@@ -0,0 +1,423 @@
+  ExecFlagEnum & exec_enum = params.set<ExecFlagEnum>("execute_on", true);
+  exec_enum.addAvailableFlags(EXEC_PRE_KERNELS);
+  exec_enum = {EXEC_PRE_KERNELS};
+  params.suppressParameter<ExecFlagEnum>("execute_on");
+  params.addRequiredParam<VariableName>(NS::pressure, "The pressure variable.");
+  params.addRequiredParam<VariableName>("u", "The x-component of velocity");
+  params.addParam<VariableName>("v", "The y-component of velocity");
+  params.addParam<VariableName>("w", "The z-component of velocity");
+  params.addClassDescription(
+      "Computes the Rhie-Chow velocity based on gathered 'a' coefficient data.");
+  return params;
+}
+
+INSFVRhieChowInterpolator::INSFVRhieChowInterpolator(const InputParameters & params)
+  : GeneralUserObject(params),
+    TaggingInterface(this),
+    BlockRestrictable(this),
+    _moose_mesh(UserObject::_subproblem.mesh()),
+    _mesh(_moose_mesh.getMesh()),
+    _dim(_moose_mesh.dimension()),
+    _vel(libMesh::n_threads()),
+    _p(dynamic_cast<INSFVPressureVariable *>(
+        &UserObject::_subproblem.getVariable(0, getParam<VariableName>(NS::pressure)))),
+    _u(dynamic_cast<INSFVVelocityVariable *>(
+        &UserObject::_subproblem.getVariable(0, getParam<VariableName>("u")))),
+    _v(isParamValid("v") ? dynamic_cast<INSFVVelocityVariable *>(
+                               &UserObject::_subproblem.getVariable(0, getParam<VariableName>("v")))
+                         : nullptr),
+    _w(isParamValid("w") ? dynamic_cast<INSFVVelocityVariable *>(
+                               &UserObject::_subproblem.getVariable(0, getParam<VariableName>("w")))
+                         : nullptr),
+    _ps(libMesh::n_threads(), nullptr),
+    _us(libMesh::n_threads(), nullptr),
+    _vs(libMesh::n_threads(), nullptr),
+    _ws(libMesh::n_threads(), nullptr),
+    _sub_ids(blockRestricted() ? blockIDs() : _moose_mesh.meshSubdomains()),
+    _sys(*getCheckedPointerParam<SystemBase *>("_sys")),
+    _example(0)
+{
+  if (!_p)
+    paramError(NS::pressure, "the pressure must be a INSFVPressureVariable.");
+
+  auto fill_container = [this](const auto & name, auto & container)
+  {
+    for (const auto tid : make_range(libMesh::n_threads()))
+    {
+      auto * const var = static_cast<MooseVariableFVReal *>(
+          &UserObject::_subproblem.getVariable(tid, getParam<VariableName>(name)));
+      container[tid] = var;
+    }
+  };
+
+  auto check_blocks = [this](const auto & var)
+  {
+    if (blockIDs() != var.blockIDs())
+      mooseError("Block restriction of interpolator user object '",
+                 this->name(),
+                 "' doesn't match the block restriction of variable '",
+                 var.name(),
+                 "'");
+  };
+
+  fill_container(NS::pressure, _ps);
+  check_blocks(*_p);
+
+  if (!_u)
+    paramError("u", "the u velocity must be an INSFVVelocityVariable.");
+  fill_container("u", _us);
+  check_blocks(*_u);
+  _var_numbers.push_back(_u->number());
```

Comment:
so for _u auxiliary variable, the kernel check later (checking that momentum kernels are momentum ROs) will pull in the kernels for another variable (with the same number, in the NL system). 

So we should skip this when _u/_v/_w are auxiliary variables.
Doesnt solve the fact that we dont have any 'a' data for them, I need to think about this

Reply:
if _a could be a functor, that would solve the problem. We could just transfer it as an auxvariable

Reply:
let's add a test with auxvariable advection so we dont break that

Reply:
So make `_a` a `CellCenteredMapFunctor` and then we can create an aux field from it for transferring from... and then in our transferred-to app, where our velocities are aux variables we don't bother with the RC data gathering because we should already have it ... ? It seems then that for ease-of-use I should have an `_a_write` `CellCenteredMapFunctor` (that I can write to in the fluid flow app), and then an `_a_read` functor that is a generic `const Moose::Functor<ADReal> &` that we read from (regardless of fluid-flow app or scalar transport app) when computing the RC velocity.

Does this sound about right?

Reply:
Seems like a cleaner plan than mine